dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/bundle/DefaultTrackerBundleService.java

```diff
@@ -278,6 +294,49 @@ private TrackerTypeReport handleEvents( Session session, TrackerBundle bundle )
         return typeReport;
     }
 
+    private TrackerTypeReport handleRelationships( Session session, TrackerBundle bundle )
+    {
+        List<Relationship> relationships = bundle.getRelationships();
+        TrackerTypeReport typeReport = new TrackerTypeReport( TrackerType.RELATIONSHIP );
+
+        relationships.forEach( o -> bundleHooks.forEach( hook -> hook.preCreate( Relationship.class, o, bundle ) ) );
+        session.flush();
+
+        for ( int idx = 0; idx < relationships.size(); idx++ )
+        {
+            Relationship relationship = relationships.get( idx );
+            org.hisp.dhis.relationship.Relationship toRelationship = relationshipTrackerConverterService
+                .from( bundle.getPreheat(), relationship );
+
+            TrackerObjectReport objectReport = new TrackerObjectReport( TrackerType.EVENT,
+                toRelationship.getUid(), idx );
+            typeReport.addObjectReport( objectReport );
+
+            Date now = new Date();
+
+            if ( bundle.getImportStrategy().isCreate() )
+            {
+                toRelationship.setCreated( now );
+            }
+
+            toRelationship.setLastUpdated( now );
+            toRelationship.setLastUpdatedBy( bundle.getUser() );
+
+            session.persist( toRelationship );
+            typeReport.getStats().incCreated();
+
+            if ( FlushMode.OBJECT == bundle.getFlushMode() )
+            {
+                session.flush();
+            }
+        }
+
+        session.flush();
```

Comment:
Lots of flushing :) why can't we delegate the flushing to hibernate?

Reply:
Removed the flushing